lib/matplotlib/ticker.py

```diff
@@ -377,19 +380,39 @@ class FuncFormatter(Formatter):
     """
     Use a user-defined function for formatting.
 
-    The function should take in two inputs (a tick value ``x`` and a
-    position ``pos``), and return a string containing the corresponding
-    tick label.
+    The function can take in at most two inputs (a required tick value ``x``
+     and an optional position ``pos``), and must return a string containing
+     the corresponding tick label.
     """
     def __init__(self, func):
         self.func = func
 
+        if not isinstance(func, types.BuiltinFunctionType):
+            self.nargs = len(inspect.signature(func).parameters)
+        elif (isinstance(getattr(func, "__self__"), str) and
+                (getattr(func, "__name__", "") == "format")):
+            #check if there's a format spec
+            self.nargs = len([(_, _, fs, _) for (_, _, fs, _)
+                              in string.Formatter().parse(func.__self__)
+                              if fs is not None])
+        else:
+            #finding argcount for other builtins is a mess
+            self.nargs = 2
+            cbook._warn_external(f"{func.__name__} is not supported "
+                                  "and may not work as expected")
+        if self.nargs not in [1, 2]:
```

Comment:
I think best practices are to always use a set for inclusion testing (`in {1, 2}`).  The performance benefit isn't a big deal here, but again it is a good habit to follow.

Reply:
This is micro-optimization on the nanosecond level, and for the present case not even a benefit
~~~
In [2]: %timeit a in [1, 2]
63 ns ± 9.71 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [3]: %timeit a in {1, 2}
73.4 ns ± 8.28 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [4]: %timeit a in (1, 2)
61.3 ns ± 7.24 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
~~~

I think the list is slightly more readable because it's used more often, and because sets and dicts use curly braces so you have to look a bit closer to realize this is a set.